src/zenml/integrations/aws/step_operators/sagemaker_step_operator.py

```diff
@@ -146,8 +155,13 @@ def launch(
                 self.name,
             )
 
-        # TODO: figure out a nice way to get the build here
-        image_name = info.config.extra[SAGEMAKER_DOCKER_IMAGE_KEY]
+        run = Client().get_pipeline_run(info.run_id)
+        assert run.build
```

Comment:
Yep just mypy, a run including a containerized orchestrator or step operator (or any other component that requires a Docker image) will always have a build associated with it, otherwise it would have failed to start the run

Reply:
I guess users could delete the build after running the pipeline but before the run finished. I think that's not something that we should handle even, at that point we might also check if they didn't delete the run while it's running